builtin.go

```diff
@@ -258,6 +258,30 @@ func Not(validator Validator) (mv *MessageValidator) {
 	return
 }
 
+// PointerValue is a composite validator factory used to create a validator, which will
+// dereference the pointer field's value and apply the given validator to the dereferenced value.
+//
+// If the field's value is not a pointer or is nil, it will return an unsupported error.
+func PointerValue[T any](validator Validator) Validator {
+	return Func(func(field *Field) Errors {
+		ptr, isOk := field.Value.(*T)
+		if !isOk {
+			return NewErrors(field.Name, ErrUnsupported, fmt.Sprintf("expected a %T but got %T", new(T), field.Value))
+		}
+
+		if ptr == nil {
+			return NewErrors(field.Name, ErrUnsupported, "expected a non-nil pointer but got nil pointer")
+		}
```

Comment:
Are there any cases where people want to customize this error message? For example, API providers typically want to notify the users for nil pointer values by returning human-readable error messages?

Reply:
I didn't use `MessageValidator` because all the possible errors returned from the function are `ErrUnsupported`s. The error message can be set for the wrapped validation function, like this:

```go
type Params struct {
	UserName *string
}

func validate(params *Params) error {
	validationErrs := v.Validate(v.Schema{
		v.F("userName", params.UserName): validatingextra.PointerValue[string](
			v.LenString(1, 10).Msg(fmt.Sprintf("The user name must be between %d and %d characters long.", 1, 10)),
		),
	})

	if len(validationErrs) > 0 {
		return validationErrs
	}

	return nil
}

```